File: app/src/main/java/com/bzh/dytt/SingleActivity.java
package com.bzh.dytt;

import android.app.Activity;
import android.content.Intent;
import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBar;
import android.view.MenuItem;

import com.bzh.dytt.home.VideoDetailPageFragment;
import com.bzh.dytt.search.SearchFragment;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.HasSupportFragmentInjector;

public class SingleActivity extends BaseActivity implements HasSupportFragmentInjector {

    public static final String TYPE = "TYPE";
    public static final String DATA = "DATA";
    @Inject
    DispatchingAndroidInjector<Fragment> fragmentInjector;

    public static void startDetailPage(Activity activity, String detailLink) {
        Intent intent = new Intent(activity, SingleActivity.class);
        intent.putExtra(TYPE, SingleType.Detail.getValue());
        intent.putExtra(DATA, detailLink);
        activity.startActivity(intent);
    }

    public static void startSearchPage(Activity activity) {
        Intent intent = new Intent(activity, SingleActivity.class);
        intent.putExtra(TYPE, SingleType.Search.getValue());
        activity.startActivity(intent);
    }

    @Override
    protected void doCreate() {
        setContentView(R.layout.activity_single);
        setupActionBar();

        Intent intent = getIntent();

        BaseFragment fragment = null;

        int type = intent.getIntExtra(TYPE, SingleType.Detail.getValue());

        if (SingleType.Detail.getValue() == type) {
            String detailLink = intent.getStringExtra(DATA);
            assert detailLink != null;
            fragment = VideoDetailPageFragment.newInstance(detailLink);
        }

        if (SingleType.Search.getValue() == type) {
            fragment = SearchFragment.newInstance();
        }

        assert fragment != null;
        getSupportFragmentManager().beginTransaction().replace(R.id.container, fragment).commit();
    }


    private void setupActionBar() {
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setHomeButtonEnabled(true);
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return fragmentInjector;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == android.R.id.home) {
            onBackPressed();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }


    private enum SingleType {
        Detail(1),
        Search(2);

        private int mValue;

        SingleType(int value) {
            mValue = value;
        }

        public int getValue() {
            return mValue;
        }
    }

}

File: app/src/main/java/com/bzh/dytt/SingleActivity.kt
package com.bzh.dytt

import android.app.Activity
import android.app.ActivityOptions
import android.content.Intent
import android.os.Build
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import android.view.MenuItem
import android.widget.ImageView
import com.bzh.dytt.base.BaseActivity
import com.bzh.dytt.base.BaseFragment
import com.bzh.dytt.ui.detail.DetailFragment
import com.bzh.dytt.ui.search.SearchFragment
import com.bzh.dytt.vo.MovieDetail
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.HasSupportFragmentInjector
import javax.inject.Inject


class SingleActivity : BaseActivity(), HasSupportFragmentInjector {

    @Inject
    lateinit var fragmentInjector: DispatchingAndroidInjector<androidx.fragment.app.Fragment>

    override fun supportFragmentInjector() = fragmentInjector

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_single)
        setupActionBar()
        setupContainerWithType()
    }

    private fun setupContainerWithType() {
        val type = intent.getSerializableExtra(TYPE)
        val fragment: BaseFragment = when (type) {
            SingleType.Detail -> {
                DetailFragment.newInstance(intent.getParcelableExtra(DATA))
            }
            else -> {
                SearchFragment.newInstance()
            }
        }
        supportFragmentManager.beginTransaction().replace(R.id.container, fragment).commitAllowingStateLoss()
    }

    private fun setupActionBar() {
        supportActionBar?.setHomeButtonEnabled(true)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        val id = item?.itemId

        if (id == android.R.id.home) {
            onBackPressed()
            return true
        }

        return super.onOptionsItemSelected(item)
    }

    enum class SingleType {
        Detail(),
        Search()
    }

    companion object {
        const val TYPE = "TYPE"
        const val DATA = "DATA"

        fun startDetailPage(activity: androidx.fragment.app.FragmentActivity?, sharedElement: ImageView, sharedKey: String, movieDetail: MovieDetail) {
            val intent = Intent(activity, SingleActivity::class.java)
            intent.putExtra(TYPE, SingleActivity.SingleType.Detail)
            intent.putExtra(DATA, movieDetail)

            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                activity?.startActivity(intent, ActivityOptions.makeSceneTransitionAnimation(activity, sharedElement, sharedKey).toBundle())
            } else {
                activity?.startActivity(intent)
            }
        }

        fun startSearchPage(activity: Activity) {
            val intent = Intent(activity, SingleActivity::class.java)
            intent.putExtra(TYPE, SingleActivity.SingleType.Search)
            activity.startActivity(intent)
        }
    }
}